adodb-pear.inc.php

```diff
@@ -358,7 +358,7 @@ function parseDSN($dsn)
 	 */
 	function assertExtension($name)
 	{
-		if (!extension_loaded($name)) {
+		if (function_exists('dl') && !extension_loaded($name)) {
```

Comment:
Probably better test for *extension_loaded* first, no need to call *function_exists* if the extension is loaded.

Reply:
I thought exactly the opposite ;-) Why bother checking if an extension is loaded if there's no way to force load it anyway.

Reply:
I guess that makes sense too. The question is, which one of the 2 conditions is the most likely to occur. But considering It would probably save about 1 nanosecond (according to the [benchmark posted by some guy on the Drupal project](https://www.drupal.org/project/drupal/issues/1942432#comment-10006777)), let's forget about it :wink: